vars/prepareDefaultValues.groovy

```diff
@@ -1,10 +1,26 @@
 import com.sap.piper.DefaultValueCache
 
+import hudson.AbortException
+
 def call(Map parameters = [:]) {
     handlePipelineStepErrors (stepName: 'prepareDefaultValues', stepParameters: parameters) {
         if(!DefaultValueCache.getInstance()) {
             Map defaultValues  = readYaml text: libraryResource('default_pipeline_environment.yml')
-            DefaultValueCache.createInstance(defaultValues)
+            Map customDefaultValues = null
+
+            def customDefaults = null
+
+            try {
+              customDefaults = libraryResource('pipeline_environment.yml')
+            } catch(AbortException e) {
+                // custom defaults file not found, that's OK the file is optional.
+            }
+
+            if( customDefaults) {
+                customDefaultValues = readYaml text: customDefaults
+            }
+            echo "CUSTOM_DEFAULT_VALUES: '${customDefaultValues}'."
+            DefaultValueCache.createInstance(defaultValues, customDefaultValues)
```

Comment:
I suggest to merge the various defaults already here and then put the consolidated defaults in the `DefaultValueCache`.

Otherwise we need to do the merging for every step call.

Reply:
see my comment above. By the way: The echo above should be removed before merging.

Reply:
+1 for merging here